### PR TITLE
Fix interface codegen

### PR DESCRIPTION
**Pull request by jhendrixMSFT**
The interfaces codegen didn't add the package name for all cases,
specifically for sequence and dictionary types.
Worked around an autorest.common bug that doesn't set the CodeModel on
sequence and dictionary types.

### CODE REVIEW

File: src/Model/MethodGo.cs [hunk 1]
```diff
@@ -125,9 +125,14 @@ public string MethodParametersSignature(bool includePkgName = false)
 
         private string ParameterTypeSig(IModelType type, bool includePkgName)
         {
-            if (includePkgName && type is CompositeTypeGo)
+            if (includePkgName)
             {
-                return $"{CodeModel.Namespace}.{type.Name}";
+                if (type.IsUserDefinedType())
+                    return $"{CodeModel.Namespace}.{type.Name}";
+                else if (type is SequenceTypeGo stg)
```
Comment (vladbarosan) [inline src/Model/MethodGo.cs]: why do we need the package prefix on these ? wont all the types be in the same package ?

Comment (jhendrixMSFT) [reply src/Model/MethodGo.cs]: no the generated interfaces are in a sub-package.

File: src/TransformerGo.cs [hunk 2]
```diff
@@ -47,6 +47,14 @@ private static void TransformPropertyTypes(CodeModelGo cmg)
                     {
                         property.Name = property.ModelType.HasInterface() ? property.ModelType.GetInterfaceName() : property.ModelType.Name.Value;
                     }
+                    if (property.ModelType is SequenceTypeGo stg)
```
Comment (vladbarosan) [inline src/TransformerGo.cs]: did we figure out why those are missing ?

File: src/Model/MethodGo.cs (hunk 1 above)
Comment (vladbarosan) [reply src/Model/MethodGo.cs]: oh ok, for the interfaces cuz they are in different packages.

File: src/TransformerGo.cs (hunk 2 above)
Comment (jhendrixMSFT) [reply src/TransformerGo.cs]: yes it's a bug in autorest.common.  I discussed with @fearthecowboy and he suggested this workaround.

File: src/Model/MethodGo.cs (hunk 1 above)
Comment (vladbarosan) [reply src/Model/MethodGo.cs]: ah vs code comments dont sink very well

**Comment by fearthecowboy** [comment]
# 🤖 AutoRest automatic publish job 🤖
## success (version: 2.1.125)

**Pull request merged**